superset/db_engine_specs/ocient.py

```diff
@@ -23,10 +23,13 @@
 from pyocient import _STPoint, _STLinestring, _STPolygon
 from ipaddress import IPv4Address, IPv6Address
 from superset import app
+from superset.models.sql_lab import Query
```

Comment:
What is this `Query` object and how is the `id` field populated?

Reply:
https://github.com/apache/superset/blob/c58dbf8b9c7c93f5c15b99557a2ec0c6294201e7/superset/models/sql_lab.py#L63

Query is an ORM in she superset backed that links to the stored queries in superset.
id is populated at query creation and is unique per query executed on superset

Reply:
hmm, how does Ocient know to correlate that superset id with the id that Ocient assigns the query internally? Has this been tested?

Reply:
My concern being that any id that Superset generates is garbage from Ocient's perspective

Reply:
Correct, that is the purpose of the cache, to tie the superset ID to a cursor object, from which we can extract the Ocient query id.

This is a workaround to the expectation that superset has that we have all the information to cancel a query upon connection, which is not true for ocient.